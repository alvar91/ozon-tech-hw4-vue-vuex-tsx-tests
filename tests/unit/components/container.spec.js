import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import Container from "@/components/container.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Container.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(Container, {
      localVue,
      slots: {
        default: "test-content",
      },
    });
  });

  it("should correctly renders markup", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should correct render slot content", () => {
    expect(wrapper.find(".container").text()).toContain("test-content");
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
