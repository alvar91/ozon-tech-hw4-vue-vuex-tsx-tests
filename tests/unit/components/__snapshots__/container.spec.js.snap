// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container.vue should correctly renders markup 1`] = `<div class="container">test-content</div>`;
